### DISCUSSION TITLE
Is it possible to pack the .config file into the packed xll?

### DISCUSSION THREAD

**Discussion by JonTheTurnip** [Excel-DNA/ExcelDna#584] [Q&A] [answered]
Hi there,

I'm currently learning ExcelDna and it's going well :)

I'm curious to know if it's possible to configure ExcelDna so that packed xll also includes the .config file?  It would be nice just to have a single xll file rather than the xll and the xll.config file.

The only reason that I have a config file in the first place is because I needed  a couple of assembly binding redirects.   My project is an SDK style project where Sdk="Microsoft.NET.Sdk.WindowsDesktop" and the target framework is net472.

Thank you!

**Comment by govert** [comment, marked as answer]
The .config file should be packed automatically if present (you should check the build output for the list of files packed).

```
1>  Task "PackExcelAddIn"
1>    PackExcelAddIn: Running PackExcelAddIn Task
1>    PackExcelAddIn: Using base add-in bin\Debug\net472\TestConfig-AddIn64.xll
1>    PackExcelAddIn:   ->  Updating resource: Type: CONFIG, Name: __MAIN__, Length: 77
1>    PackExcelAddIn:   ~~> ExternalLibrary path TestConfig.dll resolved to bin\Debug\net472\TestConfig.dll.
1>    PackExcelAddIn:   ->  Updating resource: Type: DNA, Name: __MAIN__, Length: 534
1>    PackExcelAddIn:   ->  Updating resource: Type: ASSEMBLY_LZMA, Name: TESTCONFIG, Length: 1275
1>    PackExcelAddIn: Completed Packing bin\Debug\net472\publish\TestConfig-AddIn64-packed.xll.
1>  Done executing task "PackExcelAddIn".
```

Note the "CONFIG" entry.

However, you might find that binding redirects are not applied in the packed add-in case. This should not be a problem for assemblies that are packed themselves - these are resolved purely by name - but if you have other dependencies this can be a problem. 

I should say that .config file are only applicable to the .NET Framework runtime, and not for add-ins targeting .NET 6. For these we don't have the same 'implicit' configuration file that can be loaded per add-in, and the `System,Configuration.ConfigurationManager` library and package are also somewhat problematic. So therecommendation for .NET 6 is to move to `Microsoft.Extensions.Configuration.XXX`.

**Reply by JonTheTurnip** [reply]
Thanks for the quick reply, much appreciated. 

I've just retested and it does appear that I was mistaken.  I removed the config file and the xll worked fine without it.  Apologies for wasting your time!
